server/src/main/java/org/opensearch/search/aggregations/bucket/filterrewrite/FilterRewriteOptimizationContext.java

```diff
@@ -120,7 +118,7 @@ public boolean tryOptimize(final LeafReaderContext leafCtx, final BiConsumer<Lon
             return false;
         }
 
-        Ranges ranges = getRanges(leafCtx, segmentMatchAll);
+        Ranges ranges = getRanges(leafCtx);
         if (ranges == null) return false;
```

Comment:
It looks like we return false here if ranges is null and `getRanges` should return null if `segmentMatchAll` is false anyway? So we should be avoiding the optimization anyway, wondering how this change is helping?

Reply:
Removed tryOptimize changes.